nimbus/src/client/fs_client.rs

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! A SettingssClient that uses the file-system. Used for developer ergonomics
+//! (eg, for testing against experiments which are not deployed anywhere) and
+//! for tests.
+
+use crate::error::Result;
+use crate::Experiment;
+use crate::SettingsClient;
+use std::ffi::OsStr;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+
+pub struct FileSystemClient {
+    path: PathBuf,
+}
+
+impl FileSystemClient {
+    pub fn new<P: AsRef<Path>>(path: P) -> Result<Self> {
+        Ok(Self {
+            path: path.as_ref().into(),
+        })
+    }
+}
+
+impl SettingsClient for FileSystemClient {
+    fn get_experiments_metadata(&self) -> Result<String> {
+        unimplemented!();
+    }
+
+    fn get_experiments(&self) -> Result<Vec<Experiment>> {
+        log::info!("reading experiments in {}", self.path.display());
+        let mut res = Vec::new();
+        for child in self.path.read_dir()? {
+            if let Ok(child) = child {
+                let child_path = child.path();
+                if !child_path.is_file() {
+                    log::info!("skipping {} - not a file", child_path.display());
+                }
+                // Only treat files with .json as experiments - we might end up
+                // with README etc?
+                if child_path.extension() != Some(OsStr::new(".json")) {
+                    log::info!("skipping {} - not JSON", child_path.display());
+                }
+                let file = File::open(child_path.clone())?;
+                let reader = BufReader::new(file);
+                match serde_json::from_reader::<_, Experiment>(reader) {
+                    Ok(exp) => res.push(exp),
+                    Err(e) => {
+                        log::warn!(
+                            "Malformed experiment found! File {},  Error: {}",
+                            child_path.display(),
+                            e
+                        );
```

Comment:
For the future, we may want to consider pulling this "validate the experiment JSON and emit logs/metrics about it" logic up to shared code rather than duplicating it; it's fine here for now though.